src/main/java/com/minecolonies/blocks/BlockBarrel.java

```diff
@@ -0,0 +1,216 @@
+            fillings.put(pos, MIN_FULLNESS);
+            Log.logger.info("Set Blockstate to " + worldIn.getBlockState(pos));
+            return true;
+        }
+
+        if (itemstack == null)
+        {
+            return true;
+        }
+
+        Item item = itemstack.getItem();
+
+        if (item == Items.rotten_flesh && barrelState == BARRELSTATE_FILLING)
+        {
+            Log.logger.info("item Consumed");
+
+            itemstack.stackSize--;
+
+            fullness += 1;
+            if (fullness >= MAX_FULLNESS)
+            {
+                fullness = MAX_FULLNESS;
+                worldIn.setBlockState(pos, state.withProperty(BARRELSTATE, BARRELSTATE_COMPOSTING));
+            }
+            fillings.put(pos, fullness);
+            Log.logger.info("now FULLNESS = " + fullness);
+
+            return true;
+        }
+
+        return true;
+    }
+
+    public void GetItemFromBarrel(World worldIn, EntityPlayer playerIn, ItemStack itemStack, IBlockState state, BlockPos pos)
+    {
+        final int bs = state.getValue(BARRELSTATE);
+        if (bs == 2)
+        {
+            UseBarrel(worldIn, playerIn, null, state, pos);
+        }
+    }
+
+    @Override
+    public IBlockState getStateFromMeta(final int meta)
+    {
+        return this.getDefaultState().withProperty(BARRELSTATE, meta);
+    }
+
+    @Override
+    public int getMetaFromState(final IBlockState state)
+    {
+        return state.getValue(BARRELSTATE);
+    }
+
+    /**
+     * Used to determine ambient occlusion and culling when rebuilding chunks for render.
+     *
+     * @return true
+     */
+    @Override
+    public boolean isOpaqueCube()
+    {
+        return true;
+    }
+
+    @Override
+    public void updateTick(World worldIn, BlockPos pos, IBlockState state, Random rand)
+    {
+        Log.logger.info("UpdateTick called");
+
+        int barrelState = state.getValue(BARRELSTATE);
+
+        Log.logger.info("now BARRELSTATE = " + barrelState);
+        switch (state.getValue(BARRELSTATE))
+        {
+            case BARRELSTATE_FILLING:
+                checkIfBarrelFull(worldIn, pos, state);
+                break;
+            case BARRELSTATE_COMPOSTING:
+                doBarrelCompostTick(worldIn, pos, state);
+                break;
+            case BARRELSTATE_DONE:
+                break;
+        }
+    }
+
+    private void checkIfBarrelFull(World world, BlockPos pos, IBlockState state)
+    {
+        int fullness = fillings.getOrDefault(pos, 0);
+        if (fullness >= MAX_FULLNESS)
+        {
+            Log.logger.info("Barrel is full.");
+            world.setBlockState(pos, state.withProperty(BARRELSTATE, BARRELSTATE_COMPOSTING));
+        }
+    }
+
+    private void doBarrelCompostTick(World world, BlockPos pos, IBlockState state)
+    {
+        int timer = timers.getOrDefault(pos, 0);
+        timer++;
+        if (timer >= 20)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)